components/style/properties/gecko.mako.rs

```diff
@@ -233,29 +235,51 @@ def set_gecko_property(ffi_name, expr):
     return "self.gecko.%s = %s;" % (ffi_name, expr)
 %>
 
-<%def name="impl_keyword_setter(ident, gecko_ffi_name, keyword, cast_type='u8')">
+<%def name="impl_keyword_setter(ident, gecko_ffi_name, keyword, cast_type='u8', resolver=None)">
     #[allow(non_snake_case)]
     pub fn set_${ident}(&mut self, v: longhands::${ident}::computed_value::T) {
+        % if ident != 'outline_style':
         use properties::longhands::${ident}::computed_value::T as Keyword;
+        % endif
         // FIXME(bholley): Align binary representations and ditch |match| for cast + static_asserts
         let result = match v {
             % for value in keyword.values_for('gecko'):
+                % if resolver is not None:
+                ${resolver(to_rust_ident(value))} =>
+                    structs::${keyword.gecko_constant(value)} ${keyword.maybe_cast(cast_type)},
+                % else:
                 Keyword::${to_rust_ident(value)} =>
                     structs::${keyword.gecko_constant(value)} ${keyword.maybe_cast(cast_type)},
+                % endif
+
+                % if ident == 'outline_style':
+                    % if to_rust_ident(value) == 'none':
+
+                Either::First(Auto) =>
+                    structs::${keyword.gecko_constant('solid')} ${keyword.maybe_cast(cast_type)},
+
+                    % endif
+                % endif
             % endfor
         };
         ${set_gecko_property(gecko_ffi_name, "result")}
     }
 </%def>
 
-<%def name="impl_keyword_clone(ident, gecko_ffi_name, keyword)">
+<%def name="impl_keyword_clone(ident, gecko_ffi_name, keyword, resolver=None)">
```

Comment:
While it's probably a clever idea to wrap the keyword-constant conversion, I think this adds a bit of complication to what's there already (I don't especially like having a function check for `outline-style`). Also, since this is only needed for `outline-style`, let's simply write the autogenerated code for now.